Dalamud/Game/Addon/Lifecycle/AddonEvent.cs

```diff
@@ -1,72 +1,121 @@
+    /// <see cref="AddonSetupArgs.AtkValueSpan"/> prior to the addon being created.
     /// </summary>
+    /// <seealso cref="AddonSetupArgs"/>
     PreSetup,
-
+    
     /// <summary>
-    /// Event that is fired after an addon has completed it's setup process.
+    /// An event that is fired after an addon has finished its initial setup. This event is particularly useful for
+    /// developers seeking to add custom elements to now-initialized and populated node lists, as well as reading data
+    /// placed in the AtkValues by the game during the setup process.
+    /// See <see cref="PreSetup"/> for more information.
     /// </summary>
     PostSetup,
 
     /// <summary>
-    /// Event that is fired before an addon begins update.
+    /// An event that is fired before an addon begins its update cycle via <see cref="AtkUnitBase.Update"/>. This event
+    /// is fired every frame that an update is loaded, regardless of visibility.
     /// </summary>
+    /// <seealso cref="AddonUpdateArgs"/>
     PreUpdate,
 
     /// <summary>
-    /// Event that is fired after an addon has completed update.
+    /// An event that is fired after an addon has finished its update.
+    /// See <see cref="PreUpdate"/> for more information.
     /// </summary>
     PostUpdate,
 
     /// <summary>
-    /// Event that is fired before an addon begins draw.
+    /// An event that is fired before an addon begins drawing to screen via <see cref="AtkUnitBase.Draw"/>. Unlike
+    /// <see cref="PreUpdate"/>, this event is only fired if an addon is visible or otherwise drawing to screen.
     /// </summary>
+    /// <seealso cref="AddonDrawArgs"/>
     PreDraw,
 
     /// <summary>
-    /// Event that is fired after an addon has completed draw.
+    /// An event that is fired after an addon has finished its draw to screen.
+    /// See <see cref="PreDraw"/> for more information.
     /// </summary>
     PostDraw,
 
     /// <summary>
-    /// Event that is fired before an addon is finalized.
+    /// An event that is fired immediately before an addon is finalized via <see cref="AtkUnitBase.Finalize"/> and
+    /// destroyed. After this event, the addon will destruct its UI node data as well as free any allocated memory.
+    /// This event can be used for cleanup and tracking tasks.
     /// </summary>
+    /// <remarks>
+    /// This event is <em>NOT</em> fired when the addon is being hidden, but tends to be fired when it's being properly
+    /// closed.
+    /// <br />
+    /// As this is part of the destruction process for an addon, this event does not have an associated Post event.
+    /// </remarks>
+    /// <seealso cref="AddonFinalizeArgs"/>
     PreFinalize,
     
     /// <summary>
-    /// Event that is fired before an addon begins a requested update.
+    /// An event that is fired before an addon begins a requested update via
+    /// <see cref="AtkUnitBase.OnRequestedUpdate"/>. Requested updates generally occur when the server sends new data to
```

Comment:
I know you use this explanation a lot, but you’re completely leaving out the fact that this is triggered by subscribed AtkArrayData.

Reply:
How do you suggest wording this?

Reply:
I don't know, keeping it simple?
```cs
/// <summary>
/// An event fired before an addon begins a requested update via <see cref="AtkUnitBase.OnRequestedUpdate"/>.
/// Requested updates occur when data in the subscribed AtkArrayData has changed.
/// </summary>
```